providers/defltprov.c

```diff
@@ -369,6 +369,7 @@ static const OSSL_ALGORITHM deflt_asym_cipher[] = {
 static const OSSL_ALGORITHM deflt_keymgmt[] = {
 #ifndef OPENSSL_NO_DH
     { "DH:dhKeyAgreement", "provider=default", dh_keymgmt_functions },
+    { "X9.42 DH:dhpublicnumber:DHX", "provider=default", dhx_keymgmt_functions },
```

Comment:
Hmmm. The name does really conform to our standard naming pattern:

````
ALGNAME[VERSION?][-SUBNAME[VERSION?]?][-SIZE?][-MODE?]
````

Reply:
This is the old definition...
#define SN_dhpublicnumber               "dhpublicnumber"
#define LN_dhpublicnumber               "X9.42 DH"

So I think we may be stuck with it?

Reply:
We can have the old name as an alias, but we should define a new one that does conform as the first named entry in the list. Just re-ordering might be sufficient:

````
"DHX:X9.42 DH:dhpublicnumber"
````

Or possibly, as an alternative:

````
"DH-X9.42:X9.42 DH:dhpublicnumber:DHX"
````

Reply:
I might have to change serdertest a bit to get that to work. The tag wont match the output tag I think..

Reply:
Just did the first suggestion.